base/steer/FairRootManager.cxx

```diff
@@ -204,21 +341,28 @@ TFile* FairRootManager::OpenOutFile(const char* fname)
   return OpenOutFile(fOutFile);
 }
 
-// a common implementation for Register
 //_____________________________________________________________________________
 template<typename T>
-void FairRootManager::RegisterImpl(const char* name, const char *folderName, T* obj, Bool_t toFile) {
+void FairRootManager::RegisterImpl(const char* name, const char *folderName, T* obj, Bool_t toFile)
+{
+/// a common implementation for Register
+
   FairMonitor::GetMonitor()->RecordRegister(name,folderName,toFile);
 
+  LogMessage("Going to lock for Register (1)");
+  TMCAutoLock lk(&registerMutex);
```

Comment:
What is the advantage of using `TMCAutoLock` versus a lock from the C++11 standard? In the latter case one could use a resource managing paradigm (RAII) that unlocks the code at function exit automatically. See here for a discussion:

http://en.cppreference.com/w/cpp/language/raii

Reply:
The answer is already in this thread in one of the folded review subthreads above. As far as I could see [`TMCAutoLock`](https://root.cern.ch/doc/master/TMCAutoLock_8h_source.html) is already RAII capable.